dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/datavalue/DataValueServiceTest.java

```diff
@@ -104,27 +112,35 @@ class DataValueServiceTest extends TransactionalIntegrationTest
     public void setUpTest()
         throws Exception
     {
-        deA = createDataElement( 'A' );
-        deB = createDataElement( 'B' );
-        deC = createDataElement( 'C' );
-        deD = createDataElement( 'D' );
-        dataElementService.addDataElement( deA );
-        dataElementService.addDataElement( deB );
-        dataElementService.addDataElement( deC );
-        dataElementService.addDataElement( deD );
-        peA = createPeriod( getDay( 5 ), getDay( 6 ) );
-        peB = createPeriod( getDay( 6 ), getDay( 7 ) );
-        peC = createPeriod( getDay( 7 ), getDay( 8 ) );
-        peX = createPeriod( getDay( 27 ), getDay( 28 ) );
-        ouA = createOrganisationUnit( 'A' );
-        ouB = createOrganisationUnit( 'B' );
-        ouC = createOrganisationUnit( 'C' );
-        ouD = createOrganisationUnit( 'D' );
-        organisationUnitService.addOrganisationUnit( ouA );
-        organisationUnitService.addOrganisationUnit( ouB );
-        organisationUnitService.addOrganisationUnit( ouC );
-        organisationUnitService.addOrganisationUnit( ouD );
+        setup.addDataElement( "A", withDefaultSetup );
```

Comment:
Consider offering a more opinionated service/interface which automatically creates a set of metadata, where you can specify just the number of objects you want and you get what you get. In many cases it does not matter so much exactly what the metadata is called, you just need a reasonable set of metadata as basis for tests.